diff --git a/sites/all/modules/user_relationships/user_relationships.module b/sites/all/modules/user_relationships/user_relationships.module
index 4717a42..321fc24 100644
--- a/sites/all/modules/user_relationships/user_relationships.module
+++ b/sites/all/modules/user_relationships/user_relationships.module
@@ -672,6 +672,7 @@ function user_relationships_save_relationship($relationship, $action = 'request'
       if ($relationship->approved && !$relationship_type->is_oneway) {
         // Make sure a reversed relationship entry is created if it does not
         // exist yet.
+        $relationship->rid = db_next_id(db_query('SELECT MAX(rid) FROM {user_relationships}')->fetchField());
         db_merge('user_relationships')
           ->key(array(
               'rid' => $relationship->rid,
